plan/logical_plan_builder.go

```diff
@@ -821,11 +822,43 @@ func (b *planBuilder) buildNewTableDual() LogicalPlan {
 	return dual
 }
 
+func (b *planBuilder) getTableStats(table *model.TableInfo) *statistics.Table {
+	txn, err := b.ctx.GetTxn(false)
+	if err != nil {
+		b.err = errors.Trace(err)
+		return nil
+	}
+	// mock ctx
+	if txn == nil {
+		return statistics.PseudoTable(table)
+	}
+	m := meta.NewMeta(txn)
+	statsPB, err := m.GetTableStats(table.ID)
```

Comment:
1. Use a cache with TTL.
2. When running analyze stmt or cache entry timeout, update the cache.